app/models/miq_server/worker_management/kubernetes.rb

```diff
@@ -215,9 +215,32 @@ def save_pod(pod)
 
     ch = Concurrent::Hash.new
     ch[:label_name]            = pod.metadata.labels.name
-    ch[:last_state_terminated] = pod.status.containerStatuses.any? { |cs| cs.lastState.terminated }
     ch[:container_restarts]    = pod.status.containerStatuses.sum { |cs| cs.restartCount.to_i }
 
+    ch[:last_state_terminated] = false
+    ch[:terminations] = []
```

Comment:
I chose to clear the array each time through.  The server code that checks the `current_pods` concurrent hash can possibly miss terminations or deal with things it already processed... so we might want to just monitor for termination events as mentioned 👇 .  I'm not sure how much work that is.